runtime/moonbase/src/xcm_config.rs

```diff
@@ -404,24 +418,33 @@ pub enum CurrencyId {
 	ForeignAsset(AssetId),
 	// Our local assets
 	LocalAssetReserve(AssetId),
+	// Erc20 token
+	Erc20 { contract_address: H160 },
 }
 
 impl AccountIdToCurrencyId<AccountId, CurrencyId> for Runtime {
 	fn account_to_currency_id(account: AccountId) -> Option<CurrencyId> {
-		match account {
+		Some(match account {
 			// the self-reserve currency is identified by the pallet-balances address
-			a if a == H160::from_low_u64_be(2050).into() => Some(CurrencyId::SelfReserve),
+			a if a == H160::from_low_u64_be(2050).into() => CurrencyId::SelfReserve,
 			// the rest of the currencies, by their corresponding erc20 address
-			// We distinguish by prefix, and depending on it we create either
-			// Foreign or Local
-			_ => Runtime::account_to_asset_id(account).map(|(prefix, asset_id)| {
-				if prefix == FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX.to_vec() {
-					CurrencyId::ForeignAsset(asset_id)
-				} else {
-					CurrencyId::LocalAssetReserve(asset_id)
+			_ => match Runtime::account_to_asset_id(account) {
+				// We distinguish by prefix, and depending on it we create either
+				// Foreign or Local
+				Some((prefix, asset_id)) => {
+					if prefix == FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX.to_vec() {
+						CurrencyId::ForeignAsset(asset_id)
+					} else {
+						CurrencyId::LocalAssetReserve(asset_id)
+					}
 				}
-			}),
-		}
+				// If no known prefix is identified, we consider that it's a "real" erc20 token
+				// (i.e. managed by a real smart contract)
+				None => CurrencyId::Erc20 {
+					contract_address: account.into(),
+				},
+			},
```

Comment:
This conversion was already pretty magical, and now we're adding more magic. I don't have any alternative to suggest, but I do think we need to be careful, especially with keeping this future-proof.

If nothing else, we should have some good tests around this.